builtin/providers/aws/resource_aws_instance.go

```diff
@@ -760,6 +760,10 @@ func fetchRootDeviceName(ami string, conn *ec2.EC2) (*string, error) {
 		rootDeviceName = image.BlockDeviceMappings[0].DeviceName
 	}
 
+	if rootDeviceName == nil {
+		return nil, fmt.Errorf("[WARN] Error finding Root Device Name for AMI (%s)", ami)
```

Comment:
> In the linked issue the problem was a bad AMI ID, which means we would have hit the early nil return up on L735.

Let's sync up to discuss. The issue I was debugging (and the only way I could repo the source issue) did not trigger L735; there was an image, but there was no root device matching in the lines above this. I couldn't use an AMI that didn't exist; TF returned an error before this point. I'll double check though...